core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -120,4 +121,12 @@ class TaskInfo(
   }
 
   private[spark] def timeRunning(currentTime: Long): Long = currentTime - launchTime
+
+  def duration(metrics : Option[TaskMetricsUIData]): Long = {
```

Comment:
Sorry. Looks like adding this method UIDate is better. Then you don't need to pass `metrics` to this method. Please also update https://github.com/apache/spark/blob/dbf842b7a8479f9566146192ffc04421591742d5/core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala#L890 to use this new method so that we can keep these two places always consistent.